src/api/java/com/minecolonies/coremod/colony/IColony.java

```diff
@@ -61,4 +66,27 @@
      * @return an int representing the id.
      */
     int getID();
+
+    /**
+     * returns the World the colony is in.
+     *
+     * @return the World the colony is in.
+     */
+    @Nullable
+    World getWorld();
+
+    /**
+     * Returns the buildings in the colony
+     *
+     * @return The buildings in the colony
+     */
+    @NotNull
+    ImmutableMap<BlockPos, IBuilding> getBuildings();
```

Comment:
I think this belong to a different pr, and could be merge well before dman 2.0 is ready